algebird-core/src/main/scala/com/twitter/algebird/Aggregator.scala

```diff
@@ -102,12 +102,12 @@ object Aggregator extends java.io.Serializable {
       override def append(l: T, r: F): T = appnd(l, r)
 
       override def appendAll(old: T, items: TraversableOnce[F]): T =
-        if (items.isEmpty) old else reduce(old, agg(items).get)
+        if (items.iterator.isEmpty) old else reduce(old, agg(items).get)
```

Comment:
Does this help in this case? don't collections usually have efficient implementations of `.isEmpty` that don't modify the collection?

Reply:
Hi @sritchie, yeah this one got me by surprise as well; so with 2.13 `TraversableOnce` is deprecated and synonymous to `IterableOnce`; `isEmpty` is deprecated in favor of `.iterator.isEmpty`.

With this there's no collection modification there is though an extra invocation that jvm inlining should™ take care of. Also AFAIK if a collection has a specific impl of `isEmpty` that is used by the `Iterator`.